cli/command/image/build.go

```diff
@@ -255,7 +321,14 @@ func runBuild(dockerCli *command.DockerCli, options buildOptions) error {
 
 	var body io.Reader = progress.NewProgressReader(buildCtx, progressOutput, 0, "", "Sending build context to Docker daemon")
 
-	authConfigs, _ := dockerCli.GetAllCredentials()
+	authConfigs := make(map[string]types.AuthConfig)
+	for registry := range referencedRegs {
+		conf, err := dockerCli.GetRegistryCredentials(registry)
+		if err != nil {
+			return err
+		}
+		authConfigs[registry] = conf
+	}
```

Comment:
This could be it's own function as well I think:

```go

func getAuthConfigsForBuild(dockerCli *command.Dockercli, referecedRegs []string) (map[string]types.AuthConfig, error) {
    ...
}